lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -483,10 +474,39 @@ export default class BitcoinProcessor {
    * value.
    */
   private getFirstBlockInGroup (block: number): number {
-    const groupId = this.getGroupId(block);
+    const groupId = this.getGroupIdFromBlock(block);
     return groupId * ProtocolParameters.groupSizeInBlocks;
   }
 
+  private async getStartingBlockForInitialization (): Promise<IBlockInfo | undefined> {
+    // We look at the latest data that we saved in the quantile calculator. The latest
+    // Id in there tells us which group of blocks have been processed.
+    const lastSavedGroupId = await this.quantileCalculator.getLastGroupId();
+
+    if (!lastSavedGroupId) {
+      // We are returning undefined because at the time of this refactor, the calling code
+      // is expecting an undefined for this case and having this code return the genesis
+      // block is going to cause more refactor.
```

Comment:
Consider adding a new issue to track this and referencing it in comment.

Reply:
Updated the comment with the new issue number.